fsim/upload_owner.go

```diff
@@ -121,6 +122,41 @@ func (u *UploadRequest) ProduceInfo(ctx context.Context, producer *serviceinfo.P
 	return false, false, nil
 }
 
+// sameFilesystem checks if two paths are on the same filesystem by comparing device IDs
+func sameFilesystem(path1, path2 string) (bool, error) {
+	var stat1, stat2 syscall.Stat_t
+
+	if err := syscall.Stat(path1, &stat1); err != nil {
```

Comment:
You're using `syscall.Stat` because you want to compare the Dev (uint64) field. However, this is not portable and will break Windows compatibility, at the least.

```bash
GOOS=windows go doc syscall Stat_t
doc: no symbol Stat_t in packages syscall
```

Please move `sameFilesystem` to `fsim/fs_unix.go` with the appropriate build flags for Unix-like OSes that have a `syscall.Stat` function and then create an implementation that always returns false (I guess?) under `fsim/fs_other.go`.